medical/models/medical_physician.py

```diff
@@ -25,71 +25,120 @@
 
+from datetime import datetime, date, timedelta
 
-class medical_physician_schedule_template(orm.Model):
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class MedicalPhysicianServices(orm.Model):
+    '''
+    Services provided by the Physician on a specific medical center.
+
+    A physician could have "surgeries" on one center but only
+    "general consultation" in another center,
+    or the same service with different prices for each medical center.
+    That's the reason to link this to res.partner instead of medical_physician.
+    '''
+    _name = 'medical.physician.services'
+    _inherits = {'product.product': 'product_id', }
+    _columns = {
+        'product_id': fields.many2one('product.product', 'Related Product',
+                                      required=True, ondelete='restrict',
+                                      help='Product related information for'
+                                           'Appointment Type'),
+        'physician_id': fields.many2one('medical.physician', 'Physician',
+                                        required=True, select=1,
+                                        ondelete='cascade'),
+        'service_duration': fields.selection(minutes, string='Duration'),
+    }
+
+
+class MedicalPhysicianScheduleTemplate(orm.Model):
+    '''
+    Available schedule for the Physiscian.
+
+    ie: A physiscian will be able to say, in this schedule on this days.
+
+    The objective is to show the availbles spaces for every physiscian
+    '''
     _name = 'medical.physician.schedule.template'
     _columns = {
         'physician_id': fields.many2one('medical.physician', 'Physician',
                                         required=True, select=1,
                                         ondelete='cascade'),
-        'day': fields.selection(days,
-                                string='Day', sort=False),
-        'start_hour': fields.selection(hours,
-                                       string='Hour'),
-        'start_minute': fields.selection(minutes,
-                                         string='Minute'),
-        'end_hour': fields.selection(hours,
-                                     string='Hour'),
-        'end_minute': fields.selection(minutes,
-                                       string='Minute'),
-        'duration': fields.selection(minutes,
-                                     string='Duration'),
+        'day': fields.selection(days, string='Day', sort=False),
+        'start_hour': fields.selection(hours, string='Hour'),
+        'start_minute': fields.selection(minutes, string='Minute'),
+        'end_hour': fields.selection(hours, string='Hour'),
+        'end_minute': fields.selection(minutes, string='Minute'),
+        'duration': fields.selection(minutes, string='Duration'),
     }
 
 
 class MedicalPhysician(orm.Model):
     _name = 'medical.physician'
-    _inherits = {
-        'res.users': 'user_id',
-    }
+    _inherits = {'res.users': 'user_id', }
```

Comment:
@TheCloneMaster IMO the physician should inherit from res.partner. The fact that a physician can access to the system and therefore is also a res.user is under the responsibility of the organisation which uses these addons. (one res.user = one payable user for Odoo)